lib/src/sources/cpuid_source.cc

```diff
@@ -11,8 +10,8 @@ namespace whereami { namespace sources {
 #if defined(__x86_64__) || defined(__i386__)
         // ebx is the PIC register in 32-bit environments; Don't clobber it
         asm volatile(
-            "xchgl %%ebx,%1; xor %%ebx,%%ebx; cpuid; xchgl %%ebx,%1"
-            : "=a" (result.eax), "=b" (result.ebx), "=c" (result.ecx), "=d" (result.edx)
+            "xchgl %%ebx,%k1; xor %%ebx,%%ebx; cpuid; xchgl %%ebx,%k1"
```

Comment:
I don't really understand how this works. However, would https://github.com/jedisct1/libsodium/blob/1.0.13/src/libsodium/sodium/runtime.c#L86-L101 be a sensible example to follow?

Reply:
The first implementation in that example is effectively the same as this version. The key part was the change from `=b` to `=&r`, which fixed the error on a test VM. I don't see any difference in outputs for what we're doing in this library (reading the CPU vendor ID string) when using 64-bit registers vs 32-bit, so it seems like this implementation is sufficient for all x86 platforms (for our purposes).